### ISSUE TITLE
Endpoint hosting a saga with abstract base class fails to start

### ISSUE THREAD

**Issue by yvesgoeleven** [Particular/NServiceBus#3771]
One of our customers is upgrading from v4 to v5 (5.2.14) and notices that an endpoint containing a saga which inherits from an abstract base class fails to start, as it tries to instantiate the abstract base class.

Here is the stack trace

```
Cannot create an abstract class. 
   at System.Runtime.Serialization.FormatterServices.nativeGetUninitializedObject(RuntimeType type)
   at NServiceBus.Sagas.TypeBasedSagaMetaModel.Create(Type sagaType, IEnumerable`1 availableTypes, Conventions conventions) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Sagas\TypeBasedSagaMetaModel.cs:line 69
   at NServiceBus.Sagas.TypeBasedSagaMetaModel.<>c__DisplayClass0_0.<Create>b__0(Type t) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Sagas\TypeBasedSagaMetaModel.cs:line 17
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at NServiceBus.Sagas.TypeBasedSagaMetaModel.Create(IList`1 availableTypes, Conventions conventions) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Sagas\TypeBasedSagaMetaModel.cs:line 16
   at NServiceBus.Features.Sagas.Setup(FeatureConfigurationContext context) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Sagas\Sagas.cs:line 42
   at NServiceBus.Features.Feature.SetupFeature(FeatureConfigurationContext config) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Features\Feature.cs:line 203
   at NServiceBus.Features.FeatureActivator.ActivateFeature(FeatureState featureState, IEnumerable`1 featuresToActivate, FeatureConfigurationContext context) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Features\FeatureActivator.cs:line 233
   at NServiceBus.Features.FeatureActivator.SetupFeatures(FeatureConfigurationContext context) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Features\FeatureActivator.cs:line 116
   at NServiceBus.Configure.Initialize() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Configure.cs:line 115
   at NServiceBus.Bus.Create(BusConfiguration configuration) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Bus.cs:line 21
   at NServiceBus.GenericHost.PerformConfiguration(Action`1 moreConfiguration) in c:\BuildAgent\work\a3de8759ee491634\src\NServiceBus.Hosting.Windows\GenericHost.cs:line 121
   at NServiceBus.GenericHost.Start() in c:\BuildAgent\work\a3de8759ee491634\src\NServiceBus.Hosting.Windows\GenericHost.cs:line 68
   at NServiceBus.Hosting.Windows.WindowsHost.Start() in c:\BuildAgent\work\a3de8759ee491634\src\NServiceBus.Hosting.Windows\WindowsHost.cs:line 30
   at NServiceBus.Hosting.Windows.Program.<>c__DisplayClasse.<Main>b__6(WindowsHost service) in c:\BuildAgent\work\a3de8759ee491634\src\NServiceBus.Hosting.Windows\Program.cs:line 80
   at Topshelf.Internal.ControllerDelegates`1.StartActionObject(Object obj) in c:\Projects\TopShelfForNSB\src\Topshelf\Internal\ControllerDelegates.cs:line 18
   at Topshelf.Internal.IsolatedServiceControllerWrapper`1.<>c__DisplayClass2.<set_StartAction>b__1(TService service) in c:\Projects\TopShelfForNSB\src\Topshelf\Internal\IsolatedServiceControllerWrapper.cs:line 65
   at Topshelf.Internal.ServiceController`1.<.cctor>b__1(ServiceController`1 sc) in c:\Projects\TopShelfForNSB\src\Topshelf\Internal\ServiceController.cs:line 35
   at Magnum.StateMachine.LambdaAction`1.Execute(T instance, Event event, Object parameter) in :line 0
   at Magnum.StateMachine.EventActionList`1.Execute(T stateMachine, Event event, Object parameter) in :line 0
```

@ramonsmits traced down the root cause of the problem, being: TypeBasedSagaMetaModel does not excluded abstract classes from the saga type meta model.

the following line should get an additional `&& !t.IsAbstract` check
https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Sagas/TypeBasedSagaMetaModel.cs#L22

**Comment by yvesgoeleven** [comment]
@Particular/nservicebus-maintainers

**Comment by danielmarbach** [comment]
Btw. v6 is potentially also affected

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Sagas/SagaMetadata.cs#L108

**Comment by danielmarbach** [comment]
@yvesgoeleven I pushed a fix for v6 at least

**Comment by timbussmann** [comment]
closing this as this should be fixed by #3782. Feel free to reopen.
